src/main/java/com/zufar/onlinestore/cart/api/ShoppingSessionProvider.java

```diff
@@ -29,6 +33,23 @@ public ShoppingSessionDto getByUserId(final UUID userId) throws ShoppingSessionN
             log.error("The shopping session for the user with id = {} is not found.", userId);
             throw new ShoppingSessionNotFoundException(userId);
         }
+        return shoppingSessionDtoConverter.toDto(shoppingSession); //ToDo think about return type
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.READ_COMMITTED)
+    public ShoppingSessionDto removeItemFromShoppingSession(final RemoveItemFromShoppingSessionRequest request) {
+        Integer deletedItems = shoppingSessionItemRepository.deleteShoppingSessionItemById(request.shoppingSessionItemId());
+        if (deletedItems < request.shoppingSessionItemId().size()) {
+            log.error("The list of shopping session items with ids: {} does not exist.", request.shoppingSessionItemId());
```

Comment:
Deleting the shopping session items with ids = '{}' were failed

Reply:
fixed